extra_data/reader.py

```diff
@@ -1331,6 +1584,126 @@ def __iter__(self):
             yield tid, self._assemble_data(tid)
 
 
+class AliasIndexer:
+    """Access data via aliases rather than literal names.
+
+    Created by :property:`DataCollection.alias`.
+    """
+
+    __slots__ = ('data',)  # Using slots reduces instantation time.
+
+    def __init__(self, data):
+        self.data = data
+
+    def _resolve_any_alias(self, alias):
+        try:
+            literal = self.data._aliases[alias]
+        except KeyError:
+            raise AliasError(alias) from None
+
+        return literal
+
+    def _resolve_source_alias(self, alias):
+        source = self._resolve_any_alias(alias)
+
+        if isinstance(source, tuple):
+            raise ValueError(f'{alias} not aliasing a source for this data')
+
+        return source
+
+    def __getitem__(self, aliased_item):
+        if isinstance(aliased_item, tuple) and len(aliased_item) == 2:
+            # Source alias with key literal.
+            return self.data[self._resolve_source_alias(aliased_item[0]),
+                             aliased_item[1]]
+        elif isinstance(aliased_item, str):
+            # Source or key alias.
+            return self.data[self._resolve_any_alias(aliased_item)]
+
+        raise TypeError('expected source alias or (source alias, key) tuple')
+
+    def _resolve_aliased_selection(self, selection):
+        if isinstance(selection, dict):
+            res = {self._resolve_source_alias(alias): keys
+                   for alias, keys in selection.items()}
+
+        elif isinstance(selection, Iterable):
+            res = []
+
+            for item in selection:
+                if isinstance(item, tuple) and len(item) == 2:
+                    # Source alias and literal key.
+                    item = (self._resolve_source_alias(item[0]), item[1])
+                elif isinstance(item, str):
+                    item = self._resolve_any_alias(item)
+
+                    if isinstance(item, str):
+                        # Source alias.
+                        item = (item, '*')
+
+                res.append(item)
+
+        return res
+
+    def select(self, seln_or_source_glob, key_glob='*', require_all=False):
```

Comment:
```suggestion
    def select(self, seln_or_alias, key_glob='*', require_all=False):
```

This parameter isn't a glob pattern, right? (Same for .deselect())